### PR TITLE
New Module: manage F5 BIG-IP configuration sections (network/f5/bigip_config)

### PR DESCRIPTION
**Pull request by caphrim007**
##### SUMMARY
This patch adds the bigip_config module to Ansible to support
merging SCF files, saving running configs, resetting config
and validating SCF configs

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

lib/ansible/modules/network/f5/bigip_config.py

##### ANSIBLE VERSION

```

```

##### ADDITIONAL INFORMATION
Unit tests are provided. Integration tests can be found here

https://github.com/F5Networks/f5-ansible/blob/devel/test/integration/bigip_command.yaml#L23
https://github.com/F5Networks/f5-ansible/tree/devel/test/integration/targets/bigip_command/tasks

### CODE REVIEW

File: lib/ansible/modules/network/f5/bigip_config.py [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+
+
+class ArgumentSpec(object):
+    def __init__(self):
+        self.supports_check_mode = True
+        self.argument_spec = dict(
+            reset=dict(
+                type='bool',
+                default='no',
+                choices=BOOLEANS
```
Comment (gundalow) [inline lib/ansible/modules/network/f5/bigip_config.py]: No need to specify `choices`

File: lib/ansible/modules/network/f5/bigip_config.py [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+        return result
+
+
+class ArgumentSpec(object):
+    def __init__(self):
+        self.supports_check_mode = True
+        self.argument_spec = dict(
+            reset=dict(
+                type='bool',
+                default='no',
```
Comment (gundalow) [inline lib/ansible/modules/network/f5/bigip_config.py]: I think this should be false (not a string)

File: lib/ansible/modules/network/f5/bigip_config.py [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+        valid to replace the running configuration. The running
+        configuration will not be changed.
+    choices:
+      - yes
+      - no
+    default: yes
+notes:
+  - Requires the f5-sdk Python package on the host. This is as easy as pip
+    install f5-sdk.
+  - Requires Ansible >= 2.3.
```
Comment (gundalow) [inline lib/ansible/modules/network/f5/bigip_config.py]: New modules only go into `devel`, so this will only be available in Ansible 2.4

**Comment by caphrim007** [comment]
@gundalow i dont know what it's talking about about conflicts. there were no conflicts on my branch and it rebased just fine.

**Comment by caphrim007** [comment]
@gundalow ready for review

File: lib/ansible/modules/network/f5/bigip_config.py [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+
+DOCUMENTATION = '''
+---
+module: bigip_config
+short_description: Manage BIG-IP configuration sections.
+description:
+  - Manages a BIG-IP configuration by allowing TMSH commands that
+    modify running configuration, or merge SCF formatted files intp
+    the running configuration. Additionally, this module is of
+    significant importence because it allows you to save your running
```
Comment (gundalow) [inline lib/ansible/modules/network/f5/bigip_config.py]: importance

File: lib/ansible/modules/network/f5/bigip_config.py [hunk 5]
```diff
@@ -0,0 +1,365 @@
...
+    'metadata_version': '1.0'
+}
+
+DOCUMENTATION = '''
+---
+module: bigip_config
+short_description: Manage BIG-IP configuration sections.
+description:
+  - Manages a BIG-IP configuration by allowing TMSH commands that
+    modify running configuration, or merge SCF formatted files intp
```
Comment (gundalow) [inline lib/ansible/modules/network/f5/bigip_config.py]: intp -> into?

File: lib/ansible/modules/network/f5/bigip_config.py [hunk 6]
```diff
@@ -0,0 +1,365 @@
...
+      - yes
+      - no
+    default: no
+  merge_content:
+    description:
+      - Loads the specified configuration that you want to merge into
+        the running configuration. This is equivalent to using the
+        C(tmsh) command C(load sys config from-terminal merge). If
+        you need to read configuration from a file or template, use
+        Ansible's C(file) or C(template) lookup plugins respectively.
```
Comment (gundalow) [inline lib/ansible/modules/network/f5/bigip_config.py:68]: +1 to stating this.
I wonder if an example would be useful

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/f5/bigip_config.py (hunk 6 above)
Comment (caphrim007) [reply lib/ansible/modules/network/f5/bigip_config.py:68]: fixed

File: lib/ansible/modules/network/f5/bigip_config.py (hunk 4 above)
Comment (caphrim007) [reply lib/ansible/modules/network/f5/bigip_config.py]: fixed

File: lib/ansible/modules/network/f5/bigip_config.py (hunk 5 above)
Comment (caphrim007) [reply lib/ansible/modules/network/f5/bigip_config.py]: fixed

**Review by gundalow — APPROVED** [review]

**Pull request merged**